src/xenia/base/threading_win.cc

```diff
@@ -315,8 +315,7 @@ class Win32Thread : public Win32Handle<Thread> {
   ~Win32Thread() = default;
 
   void set_name(std::string name) override {
-    AssertCallingThread();
-    xe::threading::set_name(name);
+    xe::threading::set_name(handle_, name);
```

Comment:
You sure about that? Like I said, set_name takes a thread id, and you are passing a thread handle. Are you sure that they are the same? According to the docs they are not expected to be (in that you can have multiple handles that resolve to the same thread id).